tools/SourceKit/tools/sourcekitd/bin/InProc/CodeCompletionSwiftInterop.h

```diff
@@ -0,0 +1,408 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_C_CODE_COMPLETION_H
+#define SWIFT_C_CODE_COMPLETION_H
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+/// The version constants for the SwiftCodeCompletion C API.
+/// SWIFTIDE_VERSION_MINOR should increase when there are API additions.
+/// SWIFTIDE_VERSION_MAJOR is intended for "major" source/ABI breaking changes.
+#define SWIFTIDE_VERSION_MAJOR 0
+#define SWIFTIDE_VERSION_MINOR 1
+
+#define SWIFTIDE_VERSION_ENCODE(major, minor) (((major)*10000) + ((minor)*1))
+
+#define SWIFTIDE_VERSION                                                       \
+  SWIFTIDE_VERSION_ENCODE(SWIFTIDE_VERSION_MAJOR, SWIFTIDE_VERSION_MINOR)
+
+#define SWIFTIDE_VERSION_STRINGIZE_(major, minor) #major "." #minor
+#define SWIFTIDE_VERSION_STRINGIZE(major, minor)                               \
+  SWIFTIDE_VERSION_STRINGIZE_(major, minor)
+
+#define SWIFTIDE_VERSION_STRING                                                \
+  SWIFTIDE_VERSION_STRINGIZE(SWIFTIDE_VERSION_MAJOR, SWIFTIDE_VERSION_MINOR)
+
+#ifdef __cplusplus
+#define SWIFTIDE_BEGIN_DECLS extern "C" {
+#define SWIFTIDE_END_DECLS }
+#else
+#define SWIFTIDE_BEGIN_DECLS
+#define SWIFTIDE_END_DECLS
+#endif
+
+#ifndef SWIFTIDE_PUBLIC
+#if defined(_MSC_VER)
+#define SWIFTIDE_PUBLIC __declspec(dllimport)
+#else
+#define SWIFTIDE_PUBLIC
+#endif
+#endif
+
+#ifndef __has_feature
+#define __has_feature(x) 0
+#endif
+
+#if !__has_feature(blocks)
+#error -fblocks is a requirement to use this library
+#endif
+
+#if defined(__clang__) || defined(__GNUC__)
+#define SWIFTIDE_DEPRECATED(m) __attribute__((deprecated(m)))
+#endif
+
+SWIFTIDE_BEGIN_DECLS
+
+//=== Types ---------------------------------------------------------------===//
+
+/// Global state across completions including compiler instance caching.
+typedef void *swiftide_connection_t;
+
+/// Opaque completion item handle, used to retrieve additional information that
+/// may be more expensive to compute.
+typedef void *swiftide_completion_item_t;
+
+typedef enum {
```

Comment:
Not a blocker, but this file should also get changes in https://github.com/swiftlang/sourcekit-lsp/pull/1906/commits/16f897aefa78d1d7c9e4a2fd2c5f730602b1f6d3#diff-39e84eb3460d8f1ea3f90414efa55ebfabe28d7a69b09a1cfdc3b3de7946f459